galley/pkg/config/source/priority/source.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package priority
+
+import (
+	"sync"
+
+	"istio.io/istio/galley/pkg/config/collection"
+	"istio.io/istio/galley/pkg/config/event"
+)
+
+// Source is a processor.Source implementation that combines multiple sources in priority order
+// Such that sources later in the input list take priority over sources earlier in the list
+type Source struct {
+	mu      sync.Mutex
+	started bool
+
+	inputs  []event.Source
+	handler event.Handler
+
+	eventStateMu    sync.Mutex
+	eventPriorities map[string]int
```

Comment:
it is not clear what is the ordering (i.e. 0 is highest, or lowers priority?)

Reply:
Added another comment to clarify the behavior